### PR TITLE
Issue 404 - Add source map to package

### LINKED ISSUE #404: Add source map to npm and pypi packages
Relates to plotly/dash#480

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
Fixes #404 

Relates to plotly/dash#478
https://github.com/plotly/dash-renderer/issues/104
Relates to https://github.com/plotly/dash-table/issues/284

### CODE REVIEW

File: dash_core_components/__init__.py [hunk 1]
```diff
@@ -47,6 +47,16 @@
...
+    },
+    {
+        'relative_package_path': '{}.min.js.map'.format(__name__),
+        'dev_package_path': '{}.dev.js.map'.format(__name__),
+        'external_url': (
+            'https://unpkg.com/dash-core-components@{}'
+            '/dash_core_components/dash_core_components.min.js.map'
+        ).format(__version__),
+        'namespace': 'dash_core_components',
+        'dynamic': True
```
Comment (Marc-Andre-Rivet) [inline dash_core_components/__init__.py]: Expose the map files, make them dynamic

File: dash_core_components/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dash-core-components",
-  "version": "0.40.2",
+  "version": "0.40.3",
```
Comment (Marc-Andre-Rivet) [inline dash_core_components/package.json]: bump version

File: webpack.config.js [hunk 3]
```diff
@@ -32,9 +32,7 @@ module.exports = (env, argv) => {
 
     const entry = overrides.entry || {main: './src/index.js'};
 
-    const devtool = overrides.devtool || (
-        mode === 'development' ? "eval-source-map" : 'none'
-    );
+    const devtool = overrides.devtool || 'source-map';
```
Comment (Marc-Andre-Rivet) [inline webpack.config.js]: default to source-map now -- this causes the dev bundle to increase dramatically in number of lines, even if it is smaller now -- same issue as with https://github.com/plotly/dash-renderer/pull/105#issuecomment-445050463

**Comment by Marc-Andre-Rivet** [comment]
Still need to figure out how why that *one* percy test is failing reliably.

**Comment by valentijnnieman** [comment]
Oops, you're right.

**Comment by T4rk1n** [comment]
The snapshot is too soon after starting the server, add a time.sleep between those lines:

https://github.com/plotly/dash-core-components/blob/19ac349065f315d94f1e3aee348cad74c8013328/test/test_integration.py#L817-L819

**Comment by Marc-Andre-Rivet** [comment]
Ok. Will try that and force re-run it a few times.

**Pull request merged**